MDANSE/Src/MDANSE/MolecularDynamics/Trajectory.py

```diff
@@ -54,11 +60,223 @@ def __init__(self, filename, trajectory_format=None):
+        return set(self.atom_types)
+
+    @property
+    def unique_names(self) -> set[str]:
+        """Set of unique atom labels in the current selection."""
+        if self._selection:
+            return set(self.selection_getter(self.atom_names))
+        return set(self.atom_names)
+
+    def set_transmutation(self, changed_atoms: dict[int, str]):
+        """Apply transmutation to atom types in the trajectory.
+
+        Parameters
+        ----------
+        changed_atoms : dict[int, str]
+            Substitution dictionary, as created by AtomTransmutationConfigurator
+        """
+        self._transmutation = changed_atoms
+
+    def set_selection(self, selected_indices: Sequence[int]):
+        """Apply atom selection to the atoms in the trajectory.
+
+        Parameters
+        ----------
+        selected_indices : Sequence[int]
+            Selected atom indices, output by ReusableSelection.
+        """
+        self._selection = selected_indices
+        self.selection_getter = itemgetter(*selected_indices)
+
+    def set_grouping(self, grouping_level: str):
+        """Assign the grouping level to the trajectory.
+
+        Parameters
+        ----------
+        grouping_level : str
+            Grouping level, as output by GroupingLevelConfigurator.
+        """
+        self._grouping_level = grouping_level
+
+    def get_weights(
+        self, *, prop: str | None = None,
+    ) -> tuple[dict[str, float], dict[str, float]]:
+        """Generate a dictionary of weights.
+
+        Parameters
+        ----------
+        prop : str or None, optional
+            The property to generate the weights from, if None then the
+            property set in this configurator will be used.
+
+        Returns
+        -------
+        tuple[dict[str, float], dict[str, float]]
+            The dictionary of the weights.
+
+        """
+        weights = []
+        for n_elements, atm_names, atm_elements in [
+            (self.get_natoms(), self.selection_getter(self.atom_names), self.selection_getter(self.atom_types)),
+            (
+                self.get_all_natoms(),
+                self.atom_names,
+                self.atom_types,
+            ),
+        ]:
+            w = defaultdict(float)
+            for name, element in zip(atm_names, atm_elements):
+                w[name] += self._trajectory.get_atom_property(element, prop)
+            for element, num_atoms in n_elements.items():
+                w[element] /= num_atoms
+            weights.append(w)
+
+        return tuple(weights)
+
+    def get_natoms(self) -> dict[str, int]:
+        """Count the selected atoms, per element.
+
+        Returns
+        -------
+        dict
+            A dictionary of the number of atom per element.
+
+        """
+        all_atom_names = self.atom_names
+        if self._selection:
+            return Counter(self.selection_getter(all_atom_names))
+        return Counter(all_atom_names)
```

Comment:
Any reason for the temporary name assign?
```suggestion
        if self._selection:
            return Counter(self.selection_getter(self.atom_names))
        return Counter(self.atom_names)
```